File: POO_2026_1/src/POO/Jogador.java
package POO;

import MeuPacote.Geral;
import java.util.ArrayList;

public class Jogador extends Pessoa {
  private int posicao;
  private double salario;

    public Jogador(int idPessoa, String nome, char sexo, Data dtNasc, int posicao, double salario) {
        super(idPessoa, nome, sexo, dtNasc);
        this.posicao = posicao;
        this.salario = salario;
    }

    public int getPosicao() {
        return posicao;
    }

    public void setPosicao(int posicao) {
        this.posicao = posicao;
    }

    public double getSalario() {
        return salario;
    }

    public void setSalario(double salario) {
        this.salario = salario;
    }
    
    public String ondeJoga() {
      String posicoes[] = {"XXX", "Goleiro", "Lateral-Direito", 
        "Lateral-Esquerdo", "Zagueiro", "Meio-Campo", "Atacante"};
      
      return(posicoes[posicao]);
    }
    
    public static void listarJogadores(ArrayList<Jogador> time) {
      System.out.println("Nome do Jogador.................... .idade. Onde Joga........... ...Salário....");
      System.out.println(Geral.repetir('-', 79));
       
      double totSal = 0;
      for(Jogador item: time) { // for-each
        System.out.print(item.toString()); 
        totSal = totSal + item.getSalario();
      }
      
      System.out.println(MeuPacote.Geral.repetir('-', 79));
      System.out.printf("%64s R$ %11.2f\n", "TOTAL", totSal);
    }

    // Polimorfismo de Sobreposição
    @Override
    public String toString() {
      String result;
      
      result = String.format("%s %d anos %s R$ %11.2f\n", 
                 getNome()+Geral.repetir('.', 35-getNome().length()), getDtNasc().calcularIdade(),
                 ondeJoga()+Geral.repetir('.', 20-ondeJoga().length()), salario);
      
        return result;
    }
  
}

File: CPP/POO_CPP/JogadoresDeFutebol_CPP/Jogador.cpp
#include <string>
#include "C:\Meus Documentos\Professor do DAELN\Aulas\TecProg\CPP\POO_CPP\MinhaData_CPP\Data.h"
#include "C:\Meus Documentos\Professor do DAELN\Aulas\TecProg\CPP\POO_CPP\CadastrarPessoa_CPP\Pessoa.cpp"
#include "C:\Meus Documentos\Professor do DAELN\Aulas\TecProg\CPP\Abr_14_Usando_MeuPacote_CPP\Usando_Str\Str.cpp"
#include "Jogador.h"

Jogador::Jogador(int idPessoa, string nome, char sexo, Data *dtNasc,
  int posicao, double salario): Pessoa(idPessoa, nome, sexo, dtNasc) {
  this->posicao = posicao;
  this->salario = salario;
}

Jogador::~Jogador() {
}

double Jogador::getPosicao() {
  return posicao;
}

void Jogador::setPosicao(double posicao) {
  this->posicao = posicao;
}

double Jogador::getSalario() {
  return salario;
}

void Jogador::setSalario(double salario) {
  this->salario = salario;
}

string Jogador::nomePosicao() {
  string pos[] = {"XXX", "Goleiro", "Lateral Direito",
      "Zagueiro", "Lateral Esquerdo", "Meio-Campo",
      "Atacante"};

  return(pos[posicao]);
}

string Jogador::toString() {
  char result[500];

  sprintf(result, "%s %d anos %-20s R$ %10.2f\n",
    (getNome()+repetir('.', 30-getNome().length())).c_str(),
    getDtNasc()->calcularIdade(),
    nomePosicao().c_str(),
    getSalario());

  return(result);
}
